app/models/space.rb

```diff
@@ -116,6 +116,18 @@ def self.roles
     Space::USER_ROLES.map { |r| Role.find_by_name(r) }
   end
 
+  def last_activitity
+    RecentActivity.where(owner: self).last
+  end
+
+  # Order by when the last activity in the space happened
+  scope :order_by_activity, -> {
```

Comment:
This one seems to be the best one I'm going to test it now and maybe add more tests for all the different activities.